apps/io-func-wallet-solution-backend/src/wallet-instance.ts

```diff
@@ -17,18 +18,26 @@ type AttestationService = {
 };
 
 export type WalletInstanceRepository = {
+  get: (
+    id: WalletInstance["id"],
+    userId: WalletInstance["userId"]
+  ) => TE.TaskEither<Error, WalletInstance>;
   insert: (walletInstance: WalletInstance) => TE.TaskEither<Error, void>;
 };
 
-type WalletInstanceEnvironment = {
+export type WalletInstanceEnvironment = {
   walletInstanceRepository: WalletInstanceRepository;
 };
 
-export type WalletInstance = {
-  id: NonEmptyString;
-  userId: User["id"];
-  hardwareKey: JWK;
-};
+export const WalletInstance = t.type({
+  id: NonEmptyString,
+  userId: User.props.id,
+  hardwareKey: t.record(t.string, t.unknown),
```

Comment:
Isn't `hardwareKey` supposed to be a `JWK`? You could use JwkPublicKey from: https://github.com/pagopa/io-wallet/blob/09fba457fefc299084249a697f2ffc9c9eda7874/apps/io-func-wallet-solution-backend/src/jwk.ts#L62

Reply:
thanks!

Reply:
then in `CreateWalletInstanceHandler` in `src/infra/http/handlers/create-wallet-instance.ts` it complains that `JWK` is not assignable to `JwkPublicKey`

Reply:
I believe you need an encoder/decoder to go from `JwkPublicKey` to jose's `JWK`. Although it seems strange to me because the parameters are the same. Try changing everything in `JwkPublicKey` and see if `jose` takes it for granted

Reply:
We decided to do it in another PR